metaflow/plugins/pypi/utils.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import platform
+import subprocess
+import sys
+import sysconfig
+from metaflow.exception import MetaflowException
+from metaflow._vendor.packaging.tags import (
+    compatible_tags,
+    _cpython_abis,
+    cpython_tags,
+    mac_platforms,
+)
+
+
+def conda_platform():
+    # Returns the conda platform for the Python interpreter
+    _32_bit_interpreter = sys.maxsize <= 2**32
+    if platform.system() == "Linux":
+        if _32_bit_interpreter:
+            return "linux-64"
+        else:
+            return "linux-32"
+    elif platform.system() == "Darwin":
+        if platform.machine() == "arm64":
+            return "osx-arm64"
+        elif _32_bit_interpreter:
+            return "osx-64"
+        else:
+            return "osx-32"
```

Comment:
same here. this lead to a very cryptic error when trying to launch on an x86_64 mac, due to the platform mismatch